src/utils/transaction.ts

```diff
@@ -572,6 +574,124 @@ const transaction = {
+                if (!response.success) {
+                  return reject(new Error(response.message ?? ''));
+                }
+
+                if (input.index >= response.tx.outputs.length) {
+                  return reject(new Error('Index outside of tx output array bounds'));
+                }
+                return resolve(this.convertFullNodeTxToHistoryTx(response));
+              })
+              .catch(err => reject(err));
+          });
+
+          if (!spentTx) {
+            // This should not happen since any errors should be treated already.
+            // This if statement is to ensure typing since spentTx starts as IHistoryTx | null.
+            throw new Error('Could not find the spent transaction');
+          }
+          // Update cache
+          txCache[input.hash] = cloneDeep(spentTx);
+        }
+      }
+
+      if (input.index >= spentTx.outputs.length) {
+        throw new Error(
+          `Index (${input.index}) outside of transaction output array bounds (${spentTx.outputs.length})`
+        );
+      }
+
+      const spentOut = spentTx.outputs[input.index];
+      inputs.push({
+        tx_id: input.hash,
+        index: input.index,
+        script: spentOut.script,
+        decoded: spentOut.decoded,
+        token_data: spentOut.token_data,
+        token: spentOut.token,
+        value: spentOut.value,
+      });
+    }
+
+    const tokensArray =
+      tx.version === CREATE_TOKEN_TX_VERSION
+        ? [{ uid: tx.hash }]
+        : tx.tokens.map(tk => ({ uid: tk }));
+    for (const output of tx.outputs) {
+      const script = output.parseScript(storage.config.getNetwork());
+      const out = {
+        value: output.value,
+        token_data: output.tokenData,
+        script: output.script.toString('hex'),
+        decoded: script?.toData() ?? {},
+        spent_by: null, // Cannot reconstruct this field
+      };
+      outputs.push(this.hydrateIOWithToken(out, tokensArray));
+    }
+
+    const histTx: IHistoryTx = {
+      tx_id: tx.hash,
+      signalBits: tx.signalBits,
+      version: tx.version,
+      weight: tx.weight,
+      timestamp: tx.timestamp ?? 0,
+      is_voided: false,
+      nonce: tx.nonce,
+      inputs,
+      outputs,
+      parents: tx.parents,
+      tokens: tx.tokens,
+      // The missing fields below are metadata that cannot be inferred from the
+      // Transaction instance.
+      // height, first_block
+    };
+
+    if (tx.version === CREATE_TOKEN_TX_VERSION) {
+      histTx.token_name = (tx as CreateTokenTransaction).name;
+      histTx.token_symbol = (tx as CreateTokenTransaction).symbol;
+    }
+
+    if (tx.version === NANO_CONTRACTS_VERSION) {
+      histTx.nc_id = (tx as NanoContract).id;
+      histTx.nc_method = (tx as NanoContract).method;
+      histTx.nc_args = (tx as NanoContract).args.map(a => a.toString('hex')).join('');
+      histTx.nc_pubkey = (tx as NanoContract).pubkey.toString('hex');
+      // XXX: is the nc_blueprint_id required?
```

Comment:
Did you test what comes from the ws in case of a nano contract initialize method?

Reply:
The integration tests should cover this.
But the question is because for the purpose of this PR this field is not used by the history processing method.
So it should not change the behavior when scanning for utxos, but i don't know if it changes anything related to nano.

If it does we should change this bit to make sure the data will be exact.